runtime/src/main/java/org/teiid/runtime/EmbeddedServer.java

```diff
@@ -379,15 +379,35 @@ public Object invoke(Object proxy, Method method, Object[] args)
 		}
 		
 		if (config.getCacheFactory() == null) {
-            config.setCacheFactory(new CacheFactory() {
-                @Override
-                public <K, V> Cache<K, V> get(String name) {
-                    return new LocalCache<>(name, 256);
+		    String infinispanConfig = config.getInfinispanConfigFile();
```

Comment:
I am trying to get rid of the inclusion of Infinispan directly or indirectly, put this in the hands of the developer of embedded to choose. I do not see much value in providing any reflective based impl except for testing. I feel this code needs to be in higher order frameworks like Thorntail or SB.

Reply:
> I am trying to get rid of the inclusion of Infinispan directly or indirectly, put this in the hands of the developer of embedded to choose.

You are still free to set a cachefactory any way you want.  

Infinispan is completely optional with this change.  It is also backwards compatible without a code change (just a pom change) and has an appropriate deprecation.  

My position is that it's not good to silently default to a non-concurrent cache implementation, so I'd prefer this change for now and we can refine it later.